Kubernetes/HOWTO-on-prem.md

```diff
@@ -0,0 +1,103 @@
+
+Done! Now, we also need the actual Linux Kubernetes binaries for v1.8. You can pull these from [the Kubernetes mainline](https://github.com/kubernetes/kubernetes/releases/tag/v1.8.0-rc.1), or build them from source as above, except using `K8SREPO=k8s.io/kubernetes` and the `release-1.8` branch. Copy these to `~/kube/bin` (note the extra `/bin` compared to the above path; we actual need to use these during master configuration, rather than just copying them to the Windows node).
+
+### Install CNI ###
+We also need to install the basic CNI binaries so that networking works. Download them from [here](https://github.com/containernetworking/plugins/releases) and copy them to `/opt/cni/bin/`.
+
+## Prepare the Master ##
+Copy the entire directory from [this repository](https://github.com/Microsoft/SDN/tree/master/Kubernetes/linux), which contains a lot of the setup scripts and Kubernetes files that are essential to this process. Check them all out to `~/kube/` and make the scripts executable. This entire directory will be getting mounted for a lot of the docker containers in future steps, so keep its structure the same as outlined in the guide. 
+
+We'll be creating the service cluster on the virtual subnet 11.0/16. This isn't configurable, so if you want to tweak this, it'll require manual modification.
+
+### Certificates ###
+First, prepare the certificates that will be used for nodes to communicate inthe cluster. Run `generate-certs.sh` (in the `certs/` directory), passing **your** external IP as its (only) parameter -- this is likely whatever the IP address of your `eth0` interface, or whatever:
+
+    ~/kube/certs $ ./generate-certs.sh 10.123.45.67   # example! replace
+
+### Allow Routing ###
+_This step may be optional, depending on whether or not you've configured your intended cluster subnet to be routable already_.
+
+Windows nodes will each grab a /24 from the cluster, so if your cluster CIDR is 192.168.0.0/16, a Windows node will use 192.168.0.1/24. Just run:
+
+    ~/kube $ generate-routes.sh 192.168
+
+### Prepare Manifests & Addons ###
+In the `manifest` folder, run the Python script, passing your master IP and the _full_ cluster CIDR:
+
+    $ python2 generate.py 10.123.45.67 192.168.0.0/16
+
+This will generate a set of YAML files. You should [re]move the Python script so that K8s doesn't mistake it for a manifest and cause problems.
+
+### Configure & Run Kubernetes ###
+Run the script `configure-kubectl.sh`, passing your external IP again, to create a configuration in `~/.kube/config` containing all of the certificate data and other parameters:
+
+    ~/kube $ ./configure-kubectl.sh 10.123.45.67
+
+Run the script `start-kubelet.sh`. You may need to run it with `sudo`. Then, in another shell, copy the Kubernetes configuration file to the local kubelet mount path:
+
+    $ sudo cp ~/.kube/config kubelet/

```

Comment:
I receive multiple errors upon running this script including Windows CR-LF (/r) character: command not found. Looks like script must be run with sudo for access various directories such as /var/run/Kubernetes/